src/main.rs

```diff
@@ -0,0 +1,51 @@
+use getopts::Options;
+use std::env;
+
+fn print_usage(program: &str, opts: Options) {
+    let brief = format!("Usage: {} [options]", program);
+    print!("{}", opts.usage(&brief));
+}
+
+fn parse_args(args: &[String]) -> Option<String> {
+    let program = args[0].clone();
+    let mut opts = Options::new();
+    opts.reqopt("r", "root", "outpack root path (required)", ".");
+    let matches = match opts.parse(&args[1..]) {
+        Ok(m) => { m }
+        Err(f) => {
+            print_usage(&program, opts);
+            panic!("{}", f.to_string())
+        }
+    };
+    Some(matches.opt_str("r").unwrap())
+}
+
+#[rocket::main]
```

Comment:
why does rocket get involved here?

Reply:
https://api.rocket.rs/master/rocket/attr.main.html

Tbh I was just following a tutorial example, but now I see that it should be using the #launch macro instead: https://api.rocket.rs/master/rocket/attr.launch.html

Reply:
No, ok, there was a reason for this which is that #launch won't work unless all code paths return a rocket instance.